tests/test_script_queue.py

```diff
@@ -348,6 +349,126 @@ def make_add_data(location, locationSalIndex=0):
         with self.assertRaises(salobj.AckError):
             await self.remote.cmd_showScript.set_start(salIndex=3579, timeout=STD_TIMEOUT)
 
+    async def check_add_log_level(self, log_level):
+        """Test script log level when adding a script to the script queue."""
+        await self.assert_next_queue(enabled=False, running=True)
+
+        await self.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+        await self.assert_next_queue(enabled=True, running=True)
+
+        # pause the queue so we know what to expect of queue state
+        await self.remote.cmd_pause.start(timeout=START_TIMEOUT)
+        await self.assert_next_queue(running=False)
+
+        async with salobj.Remote(domain=self.queue.domain, name="Script", index=I0) as script_remote:
+            await self.remote.cmd_add.set_start(
+                logLevel=log_level,
+                isStandard=False,
+                path="script1",
+                config="",
+                location=Location.LAST,
+                descr="test_add_log_level",
+                timeout=START_TIMEOUT)
+            await self.assert_next_queue(running=False, salIndices=[I0])
+
+            script_state = await script_remote.evt_state.next(flush=False, timeout=START_TIMEOUT)
+            self.assertEqual(script_state.state, ScriptState.UNCONFIGURED)
+            script_state = await script_remote.evt_state.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(script_state.state, ScriptState.CONFIGURED)
+
+            # check initial log level
+            data = await script_remote.evt_logLevel.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(data.level, logging.INFO)
+
+            # if log_level != 0 check final log level
+            # else check that no second log level was output
+            if log_level != 0:
+                data = await script_remote.evt_logLevel.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(data.level, log_level)
+            else:
+                with self.assertRaises(asyncio.TimeoutError):
+                    await script_remote.evt_logLevel.next(flush=False, timeout=0.01)
```

Comment:
Is that timeout long enough that you're always sure the "failed" case will trigger in time?

Reply:
That is a good question. To be completely safe I'd probably have to wait STD_TIMEOUT seconds, but that would really slow the test down. Under normal circumstances STD_TIMEOUT is gross overkill, but the way tests are run seems to slow some things down, especially code coverage.

That fact suggests 0.1 seconds might be reasonable. Or I could drop the test or wait STD_TIMEOUT if you think that's better.

Reply:
Would 1s be better than 0.1s? 0.1s "feels" short to me, but I don't actually know if it's all that short on an await timescale.